src/test/java/com/endava/internship/collections/StudentSetTest.java

```diff
@@ -0,0 +1,241 @@
+
+import org.junit.jupiter.api.*;
+
+import static org.assertj.core.api.Assertions.*;
+import static org.junit.jupiter.api.Assertions.*;
+import java.util.*;
+import java.time.LocalDate;
+
+public class StudentSetTest {
+    StudentSet set;
+
+
+    @BeforeEach
+    void setUpStudentSet(){
+        set = new StudentSet();
+    }
+
+    @Test
+    void addElementShouldReturnTrue(){
+        Student st = new Student("John Doe", LocalDate.of(2000, 5, 12), "");
+        assertTrue(set.add(st));
+    }
+
+    @Test
+    void addDuplicatedNullToSetShouldAllowASingleNullElement(){
+        set.add(null);
+        set.add(null);
+        assertThat(set).containsNull();
+    }
+
+    @Test
+    void addDublicatedObjectToSetShouldNotBeAdded(){
+        Student st = new Student("John Doe", LocalDate.of(2000, 5, 12), "");
+        set.add(st);
+        assertAll(
+                () -> assertFalse(set.add(st)),
+                () -> assertThat(set).containsOnlyOnce(st)
+        );
+    }
+
+    @Test
+    void addNullToSetShouldBeAdded(){
+        assertTrue(set.add(null));
+    }
+
+    @Test
+    void setShouldBeEmptyIfNoContainsElements(){
+        assertThat(set).isEmpty();
+    }
+
+    @Test
+    void sizeShouldBe0IfNoContainsElements(){
+        assertEquals(0, set.size());
+    }
+
+    @Test
+    void ifElementIsContainedItShouldReturnTrue(){
+        Student st = new Student("John Doe", LocalDate.of(2000, 5, 12), "");
+        set.add(st);
+        assertTrue(set.contains(st));
+    }
+
+    @Test
+    void ifElementIsNotContainedItShouldReturnFalse(){
+        Student st = new Student("John Doe", LocalDate.of(2000, 5, 12), "");
+        assertFalse(set.contains(st));
+    }
+
+    @Test
+    void setShouldBeClearedIfContainsElements(){
+        set.add(new Student("John Doe", LocalDate.of(2000, 5, 12), ""));
+        set.clear();
+
+        assertAll(
+                () -> assertThat(set).isEmpty(),
+                () -> assertThat(set).hasSize(0)
+        );
+    }
+
+    @Test
+    void elementShoulBeRemovedFromSetIfItIsContained(){
+        Student st = new Student("John Doe", LocalDate.of(2000, 5, 12), "");
+        set.add(st);
+
+        assertAll(
+                () -> assertTrue(set.remove(st)),
+                () -> assertTrue(!set.contains(st))
+        );
+    }
+
+    @Test
+    void elementShouldBeNotRemovedFromSetIfItIsNotContained(){
+        Student st = new Student("John Doe", LocalDate.of(2000, 5, 12), "");
+        assertFalse(set.remove(st));
+    }
+
+    @Test
+    void removeNullElementThatIsNotContainedShouldReturnFalse(){
+        assertFalse(set.remove(null));
+    }
+
+    @Test
+    void nullElementShouldBeRemovedIfItIsContained(){
+        set.add(null);
+
+        set.remove(null);
+        assertFalse(set.contains(null));
+    }
+
+    @Test
+    void methodShouldReturnIteratorThatContainsNotNullElementsFromTheSet(){
+        Student st1 = new Student("John Doe", LocalDate.of(2000, 5, 12), "");
```

Comment:
move stundents to constants, and rename st1 to JOHN

Reply:
Fixed test